pkg/abi/inputparsing.go

```diff
@@ -128,68 +130,49 @@ func getFloat64IfConvertible(v interface{}) (float64, bool) {
 // with a focus on those generated by the result of an Unmarshal using Go's default
 // unmarshalling.
 func getIntegerFromInterface(ctx context.Context, desc string, v interface{}) (*big.Int, error) {
-	i := new(big.Int)
 	switch vt := v.(type) {
+	case json.Number:
+		i, err := ethtypes.BigIntegerFromString(ctx, vt.String())
+		if err != nil {
+			return nil, i18n.WrapError(ctx, err, signermsgs.MsgInvalidIntegerABIInput, vt, v, desc)
+		}
+		return i, nil
 	case string:
-		// We use Go's default '0' base integer parsing, where `0x` means hex,
-		// no prefix means decimal etc.
-		i, ok := i.SetString(vt, 0)
-		if !ok {
-			f, _, err := big.ParseFloat(vt, 10, 256, big.ToNearestEven)
-			if err != nil {
-				return nil, i18n.NewError(ctx, signermsgs.MsgInvalidIntegerABIInput, vt, v, desc)
-			}
-			i, accuracy := f.Int(i)
-			if accuracy != big.Exact {
-				// If we weren't able to decode without losing precision, return an error
-				return nil, i18n.NewError(ctx, signermsgs.MsgInvalidIntegerABIInput, vt, v, desc)
-			}
-
-			return i, nil
+		i, err := ethtypes.BigIntegerFromString(ctx, vt)
+		if err != nil {
+			return nil, i18n.WrapError(ctx, err, signermsgs.MsgInvalidIntegerABIInput, vt, v, desc)
 		}
 		return i, nil
 	case *big.Float:
-		i, _ := vt.Int(i)
+		i, _ := vt.Int(nil)
```

Comment:
Why is this nil? 
```suggestion
		i, _ := vt.Int(vt)
```

Reply:
Not sure how this didn't fail the unit test 😮

Reply:
This code is correct @EnriqueL8 - could you let me know what you thing needs changing?

Reply:
You are spot on I understood that wrong, for reference comment above the function 
`func (x *Float) Int(z *Int)) (*Int, Accuracy)`
``` 
// If a non-nil *[Int] argument z is provided, [Int] stores
// the result in z instead of allocating a new [Int]
```